app/src/main/java/org/simple/clinic/search/results/PatientSearchResultsScreen.kt

```diff
@@ -75,13 +77,13 @@ class PatientSearchResultsScreen(context: Context, attrs: AttributeSet) : Relati
     return searchResultsView
         .upstreamUiEvents
         .ofType<RegisterNewPatient>()
-        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.searchPatientInput)) }
+        .map { PatientSearchResultRegisterNewPatient(extractPatientName(it.criteria)) }
   }
 
-  private fun extractPatientName(searchPatientInput: SearchPatientInput): String {
-    return when (searchPatientInput) {
-      is SearchPatientInput.Name -> searchPatientInput.searchText
-      is SearchPatientInput.PhoneNumber -> TODO("not yet implemented")
+  private fun extractPatientName(criteria: PatientSearchCriteria): String {
```

Comment:
Rename this appropriately.

Reply:
This was a temporary "fix" because we needed to pass in a name to the `RegisterNewPatient` event. This entire function will be removed once https://www.pivotaltracker.com/story/show/167343744 is done. I'd rather not rename a function which will be removed in an upcoming PR.

I have added a `TODO` that explains the same.